src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -258,20 +258,47 @@ public Migration migrated(HttpServletRequest request) {
 	 * 
 	 * @return
 	 */
-	private String DEVELOPER_TOKEN = "";
-	private static final int MAX_DEPTH = 1;
-
 	@RequestMapping("/box/folders")
-	public String getBoxFolders() {
-
+	public void getBoxFolders(HttpServletRequest request, HttpServletResponse response) {
+		
+		String boxClientId = env.getProperty("box_client_id");
+		String boxClientSecret = env.getProperty("box_client_secret");
+		String boxClientRedirectUri = env.getProperty("box_client_redirect_uri") + "/box/folders_authorized";
+		
+		// need to have all Box app configurations
+		if (boxClientId == null || boxClientSecret == null || boxClientRedirectUri == null)
+		{
+			log.error("Missing box integration parameters");
+			return;
+		}
+		String remoteUserEmail = request.getRemoteUser();
+		
+		// go to authentication screen
+		String boxAPIUrl = env.getProperty("box_api_url");
+		BoxUtils.authenticate(boxAPIUrl, boxClientId, boxClientRedirectUri, remoteUserEmail, response);
```

Comment:
Upon calling BoxUrils.authenticate, user is prompted with an log in page to Box ( if he has not logged in yet), and an authorization page to give read/write permission of his Box account to the CPM project. Once authorized, the response (authcode) will be embedded inside the callback url